Adjust navbar

reecebisel opened the pull request.

reecebisel committed:
initial commit

reecebisel committed:
have layouts/application use slim instead of erb

reecebisel committed:
add devise to gemfile

reecebisel committed:
install devise. change application layout to slim

reecebisel committed:
add welcomes controller

reecebisel committed:
Merge pull request #1 from reecebisel/devise

Devise

reecebisel committed:
add devise model

reecebisel committed:
Merge pull request #2 from reecebisel/devise

add devise model

reecebisel committed:
add minitest-rails to gemfile

reecebisel committed:
Merge pull request #3 from reecebisel/minitest

add minitest-rails to gemfile

reecebisel committed:
migration for article model

reecebisel committed:
add validations/tests for articles

reecebisel committed:
add databasecleaner to clean things when needed

reecebisel committed:
Merge pull request #4 from reecebisel/articles

Articles

reecebisel committed:
add bootstrap to gemfile and set it up

reecebisel committed:
add petergate to gemfile

reecebisel committed:
get views working for article actions new/index

reecebisel committed:
add new/create/index actions for articles

reecebisel committed:
update index views to use new class name

reecebisel committed:
write scope for home page

reecebisel committed:
get edit update actions working

reecebisel committed:
add show view

reecebisel committed:
add in petergate model migration

reecebisel committed:
add delete article action

reecebisel committed:
Merge pull request #5 from reecebisel/article_actions

Article actions

reecebisel committed:
add faker/factory_girl

reecebisel committed:
replace byebug w/ pry-byebug

reecebisel committed:
write controller test for articles

reecebisel committed:
update controller to pass tests

reecebisel committed:
Merge pull request #6 from reecebisel/test_articles

Test articles

reecebisel committed:
add/convert devise views to use slim

reecebisel committed:
fix types in views

reecebisel committed:
remove uneeded test folders

reecebisel committed:
Merge pull request #7 from reecebisel/convert_devise_views

Convert devise views

reecebisel committed:
fix error with registrations/edit and adjust style for navbar

reecebisel commented:
Hey, this pull request is using my last branch instead of my master branch. They are at the same codewise I just submitted the wrong branch name.

elorest commented on app/controllers/articles_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class ArticlesController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
```
This line is unnecessary as petergate will handle all of that.

reecebisel replied:
sweet. i'll remove that

elorest commented on app/models/article.rb:
```diff
@@ -0,0 +1,11 @@
+  validates :content, presence: true
+  validates :category, presence: true
+
+  scope :first_three_from_each_category, -> do 
```
Love that you wrote a scope. This is really inefficient though as it loads everything into memory and then groups it.

reecebisel replied:
my refactor doesn't load everything into scope, but it does hit the DB four times to load things correctly. I'm not in love with this solution, but it seems slightly better than loading everything into memory

elorest reviewed (commented):
Looks good. Checkout a couple of my comments.

reecebisel committed:
refactor scope

I have some concerns with my solution as it hits the db 4 times for the
query. It does avoid loading everything into memory but I don't love
it

reecebisel committed:
remove redundant line from atricles controller
